koku/koku/trino_database.py

```diff
@@ -78,6 +84,91 @@ def query_id(self) -> t.Optional[str]:
         return self._trino_error.query_id
 
 
+class TrinoNoSuchKeyException(Exception):
+    """Custom exception for NoSuchKey errors"""
+
+    def __init__(self, message, query_id=None, error_code=None):
+        super().__init__(message)
```

Comment:
Don't worry about using the parent class method. Store the message on a propery for more accurate retrieval instead of having to parse an array of args to get the message.
```suggestion
    def __init__(self, message, query_id=None, error_code=None):
        self.message = message
```